src/thunks/searchThunk.ts

```diff
@@ -1,74 +1,99 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
 import { createAsyncThunk } from '@reduxjs/toolkit';
 
 import { treeherderBaseURL } from '../common/constants';
 import type { APIPushResponse } from '../types/api';
-import type { Repository, Revision } from '../types/state';
+import type { Repository, RevisionsList, InputType } from '../types/state';
+
+interface FetchDataArgs {
+  repository: Repository['name'];
+  searchType: InputType;
+}
+
+interface FetchDataArgsIDEmail {
+  repository: Repository['name'];
+  search: string;
+  searchType: InputType;
+}
+
+interface FetchDataMeta {
+  error: string;
+  searchType: InputType;
+}
```

Comment:
I think you don't need to reject with the `searchType`, because in the reducer you use `meta.arg` directly. So you could reject with the error directly, like before.

Reply:
Ah, gotcha.